src/pipeline/add_pipeline.rs

```diff
@@ -0,0 +1,99 @@
+use crate::field::Fp31;
+use crate::pipeline::PipelineExt;
+use crate::shamir::{LagrangePolynomial, SecretSharing, Share};
+use futures::stream::StreamExt;
+use futures::Stream;
+use std::num::NonZeroU8;
+
+pub struct Leader {
+    k: NonZeroU8,
+}
+
+impl Leader {
+    #[allow(dead_code)]
+    pub fn run(
+        &self,
+        source: impl Stream<Item = (Share<Fp31>, Share<Fp31>)>,
```

Comment:
I am not sure how practical this is, given that leader here learns the secret. For addition I would expect it to be local and no communication required between helpers, they can just sum up (a[i] + b[i]) and reveal their shares when asked. You may want to implement multiplication where things get a little tricky because it requires bidirectional communication between helpers and leader

Reply:
at some point you will need to reveal the sum, right? perhaps an external party will collect the 3 shares and sum, and I can model it that way if it helps. but for the purposes of this prototype, I thought letting the leader sum and output the final value would be ok.

that being said, I can work on multiplication next, tho this same question will come up in that context